matrixprofile/preprocess.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+range = getattr(__builtins__, 'xrange', range)
+# end of py2 compatability boilerplate
+
+# Third-party imports
+import numpy as np
+
+# Project imports
+from matrixprofile import core
+
+
+def validate_preprocess_kwargs(preprocessing_kwargs):
+    """
+    Tests the arguments of preprocess function and raises errors for invalid arguments.
+
+    Parameters
+    ----------
+    preprocessing_kwargs : dict-like or None or False
+        A dictionary object to store keyword arguments for the preprocess function.
+        It can also be None/False/{}/"".
+
+    Returns
+    -------
+    valid_kwargs : dict-like or None
+        The valid keyword arguments for the preprocess function.
+        Returns None if the input preprocessing_kwargs is None/False/{}/"".
+
+    Raises
+    ------
+    ValueError
+        If preprocessing_kwargs is not dict-like or None.
+        If gets invalid key(s) for preprocessing_kwargs.
+
+    """
+    if preprocessing_kwargs != None and preprocessing_kwargs:
+
+        valid_preprocessing_kwargs_keys = {'window', 'impute_method', 'impute_direction', 'add_noise'}
+
+        if isinstance(preprocessing_kwargs,dict) == False:
+            raise ValueError("The parameter 'preprocessing_kwargs' is not dict like!")
+
+        elif set(preprocessing_kwargs.keys()).issubset(valid_preprocessing_kwargs_keys):
+            window = 4
+            impute_method = 'mean'
+            impute_direction = 'forward'
+            add_noise = True
+
```

Comment:
You need to ensure that the values for window, impute_method, impute_direction and add_noise are valid. Right now you are only checking if they are available or not.